QualifiedTeachersApi/src/QualifiedTeachersApi/Jobs/BatchSendQtsAwardedEmailsJob.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System.Transactions;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using QualifiedTeachersApi.DataStore.Crm;
+using QualifiedTeachersApi.DataStore.Sql;
+using QualifiedTeachersApi.DataStore.Sql.Models;
+using QualifiedTeachersApi.Jobs.Scheduling;
+
+namespace QualifiedTeachersApi.Jobs;
+
+public class BatchSendQtsAwardedEmailsJob
+{
+    private readonly BatchSendQtsAwardedEmailsJobOptions _batchSendQtsAwardedEmailsJobOptions;
+    private readonly DqtContext _dbContext;
+    private readonly IDataverseAdapter _dataverseAdapter;
+    private readonly IBackgroundJobScheduler _backgroundJobScheduler;
+    private readonly IClock _clock;
+
+    public BatchSendQtsAwardedEmailsJob(
+        IOptions<BatchSendQtsAwardedEmailsJobOptions> batchSendQtsAwardedEmailsJobOptions,
+        DqtContext dbContext,
+        IDataverseAdapter dataverseAdapter,
+        IBackgroundJobScheduler backgroundJobScheduler,
+        IClock clock)
+    {
+        _batchSendQtsAwardedEmailsJobOptions = batchSendQtsAwardedEmailsJobOptions.Value;
+        _dbContext = dbContext;
+        _dataverseAdapter = dataverseAdapter;
+        _backgroundJobScheduler = backgroundJobScheduler;
+        _clock = clock;
+    }
+
+    public async Task Execute(CancellationToken cancellationToken)
+    {
+        var lastAwardedToUtc = _batchSendQtsAwardedEmailsJobOptions.InitialLastAwardedToUtc;
+        var lastExecutedJob = await _dbContext.QtsAwardedEmailsJobs.OrderBy(j => j.ExecutedUtc).LastOrDefaultAsync();
+        if (lastExecutedJob != null)
+        {
+            lastAwardedToUtc = lastExecutedJob.AwardedToUtc;
+        }
+
+        // Look for new QTS awards up to the end of the day the configurable amount of days ago to provide a delay between award being given and email being sent.
+        var awardedToDayUtc = _clock.Today.AddDays(-_batchSendQtsAwardedEmailsJobOptions.EmailDelayDays).ToDateTime();
+        var awardedToUtc = new DateTime(awardedToDayUtc.Year, awardedToDayUtc.Month, awardedToDayUtc.Day, 23, 59, 59, DateTimeKind.Utc);
```

Comment:
As mentioned above, I think using 'less than' rather than 'less than or equal to' in the query may make this a little easier to follow.

Reply:
Ok so up to but not including

Reply:
Yes, keen to avoid the adding/subtracting one second if possible.

Reply:
Done